corehq/apps/users/models.py

```diff
@@ -3123,9 +3126,11 @@ def deactivate_mobile_workers(cls, domain, date_deactivation):
         :param domain: String - domain name
         :param date_deactivation: Date
         """
-        for trigger in cls.objects.filter(
+        trigger_query = cls.objects.filter(
             domain=domain,
             deactivate_after__lte=date_deactivation
-        ):
-            auto_deactivate_commcare_user(trigger.user_id, trigger.domain)
-            trigger.delete()
+        )
+        user_ids = trigger_query.values_list('user_id', flat=True)
+        for chunked_ids in chunked(user_ids, 100):
+            bulk_auto_deactivate_commcare_users(chunked_ids, domain)
```

Comment:
is it really duplicated if we actually need the chunked ids to filter the `DeactivateMobileWorkerTrigger`s?

Reply:
I didn't understand your response. Also, why are users not deleted over in corehq/apps/enterprise/models.py (like they are here)?

Maybe a better question is why are we deleting users here (next line below)? Previously I made a suggestion to improve that delete query without really thinking about why users are being deleted. Probably should have asked that first.

Reply:
@millerdev we need the `chunked_ids` in the deletion of the triggers below, so we cannot fully deduplicate the code since that also needs to be in the loop?